tests/virt/backup_test.py

```diff
@@ -362,6 +363,90 @@ def test_start_stop_backup_engine_scratch_disks(tmpdir):
         assert drive.scratch_disk is None
 
 
+def test_start_stop_backup_engine_scratch_disks_with_scratch_ids(tmpdir):
+    vm = FakeVm()
+    dom = FakeDomainAdapter()
+
+    socket = backup.socket_path(BACKUP_1_ID)
+    scratch1 = create_scratch_disk(tmpdir, "scratch1")
+    scratch2 = create_scratch_disk(tmpdir, "scratch2")
+
+    fake_disks = create_fake_disks(vm)
+
+    # Engine sends scratch disk IDs.
+    scratch_disk_domain_uuid = make_uuid()
+    scratch_disk_image_uuid = make_uuid()
+    scratch_disk_volume_uuid = make_uuid()
+
+    # Block based scratch disk for block based disk.
+    fake_disks[0]['scratch_disk'] = {
+        'path': scratch1,
+        'type': DISK_TYPE.BLOCK,
+        'domainID': scratch_disk_domain_uuid,
```

Comment:
You can use make_uuid() to create new uuid

Reply:
Done.
Inlined the 3 variables.